docs/core/accounts.md

```diff
@@ -1,177 +1,143 @@
+</Callout>
+
+## Program Account
+
+On Solana, “smart contracts” are referred to as programs. A program is an
+account that stores executable code and is indicated by an “executable” flag
+that is set to true.
+
+When new programs are
+[deployed](https://github.com/solana-labs/solana/blob/master/programs/bpf_loader/src/lib.rs#L498)
+on Solana, technically two separate accounts are created.
+
+- **Program Account**: The main account representing an on-chain program. This
+  account stores the address of an executable data account (which stores the
+  compiled program code) and the update authority for the program (address
+  authorized to make changes to the program).
+- **Program Executable Data Account**: An account that stores the executable
+  byte code of the program.
+
+<Callout>
+  The address of the "Program Account" is commonly referred to as the “Program
+  ID”, which is used to invoke the program.
+</Callout>
+
+For example, here are links to Solana Explorer for the Token Extensions
+[Program Account](https://explorer.solana.com/address/TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb)
+and its corresponding
+[Program Executable Data Account](https://explorer.solana.com/address/DoU57AYuPFu2QU514RktNPG22QhApEjnKxnBcu4BHDTY).
+
+![Program and Executable Data Accounts](/assets/docs/core/accounts/program-account-expanded.svg)
+
+For simplicity, you can think of the "Program Account" as the program itself.
+
+![Program Account](/assets/docs/core/accounts/program-account-simple.svg)
+
+Program accounts are owned by the
+[BPF Loader](https://github.com/solana-labs/solana/tree/master/programs/bpf_loader/src)
+program, another one of Solana's “Native Programs”. There are multiple versions
+of the BPF Loader program, but at a high level, it's simply the program that
+owns other programs (with the exception of Native Programs).
+
+## Data Account
+
+Solana programs are "stateless", meaning that program accounts only store the
+program's executable byte code. To store and modify additional data, new
+accounts must be created. These accounts are commonly referred to as “data
+accounts”.
+
+Data accounts can be used to store any data as defined by the owner program.
+
+![Data Account](/assets/docs/core/accounts/data-account.svg)
+
+Note that only the [System Program](/docs/core/accounts#system-program) can
+create new accounts. Once the System Program creates an account, it can then
+transfer ownership of the new account to another program.
+
+In other words, creating a data account for a custom program requires two steps:
+
+1. Invoke the System Program to create an account, which then transfers
+   ownership to a custom program
+2. Invoke the custom program, which now owns the account, to then initialize the
+   account data as defined in the program logic
+
+This data account creation process is often abstracted as a single step, but
+it's helpful to understand the underlying process.
```

Comment:
Could we get nice conclusions to summarize the learnings on this page? Highlight the most important bits at the very least.

Reply:
Not sure if it makes sense to add summary at end of each page for docs since it's not likely going to be read from top to bottom. Maybe keypoint bullet points at top of page?

Reply:
They're docs. Are they not supposed to be read from top to bottom?

We can do the keypoint bullets at the top. That's what we did with cookbook. We also did summaries to drive in the points.

Reply:
What I meant is that readers might not necessarily always reach the bottom of the page (for example if they're just scanning through quickly for a specific topic)

Reply:
added key points to top of pages